### ISSUE TITLE
Instructions for producing a signed app

### ISSUE THREAD

**Issue by isoboroff** [jimeh/build-emacs-for-macos#64]
The README says that code sign instructions are coming soon.  Even a pointer to what you have to set up and pass to the `go run ./cmd/emacs-builder package` would be great!

I'm building on an M1 machine, otherwise I would slack off and use the Homebrew cask.

**Comment by jimeh** [comment]
Apologies, I haven't gotten around to properly documenting it. It's a bit involved, but below are some very brief instructions that hopefully will help.

First, you'll need a paid for Apple Developer account, with a application signing key available on your machine, and then set a few environment variables to relevant values, and run the emacsbuilder `package` command with the `--sign` flag.

You can view the help info for the package command by running `go run ./cmd/emacs-builder package --help`. The environment variables needed are `AC_USERNAME`, `AC_PASSWORD`, `AC_PROVIDER`, and `AC_SIGN_IDENTITY`. The help output has a brief description of each.

Then you should be able to sign, package and notarize a folder containing a `Emacs.app` bundle to a `*.dmg` by running:

```
go run ./cmd/emacs-builder package --sign /path/to/folder
```

If it helps, the relevant part in my GitHub Actions setup for nightly builds is [here](https://github.com/jimeh/emacs-builds/blob/16087a80d0676e696fd1b1ed377daee683ef3a25/.github/workflows/_build.yml?_pjax=%23js-repo-pjax-container%3Afirst-of-type%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%3Afirst-of-type%2C%20%5Bdata-pjax-container%5D%3Afirst-of-type#L225-L233). The only thing the `package` command pulls out of the passed in `build-plan.yml` file is the source directory to package.

**Comment by isoboroff** [comment]
Thanks.  The symptom is that the app won't launch, and I get a crash report SIGKILL (Code Signature Invalid), so I can't even allow the app in the security and privacy settings.

**Comment by jimeh** [comment]
Right, yeah. It seems M1 machines refuse to run unsigned code altogether. On Intel-based Macs it seems to be aware of what binaries were built locally, so they're automatically trusted.

I don't have a M1 machine myself yet though, so my knowledge is mostly second hand right now.

**Comment by isoboroff** [comment]
Indeed running Contents/MacOS/bin/emacs from the command line also gets killed.  However, a hand-coded hello-world.c compiled with /usr/bin/gcc works ok.

I need to go google more so I can understand where the boundary is.

**Comment by ccqpein** [comment]
@isoboroff @jimeh I just met the same issue yesterday, and I use the same command to fix the issue. Instead of manually codesigning the `.app`, is it possible that the script signs it for us?

**Comment by jimeh** [comment]
I had not realized ad-hoc signing without any paid developer certificates was possible. I'll have a look at automatically doing that on non-Intel Macs when I finally get an Apple Silicon machine later this month.
